feat: add web modal dialog criteria

marshcooke opened the pull request:
Web Modal Dialog criteria review checklist:

- [ ]  Run the project
- [ ]  Cross compare the old & new side by side, note anything missing
- [ ]  Note if anything is styled oddly or missing or broken
- [ ]  Read the new criteria file, note anywhere in the content where edits or changes are needed

TODO needed: get the modal dialog example working

marshcooke commented on public/content/documentation/web/component/modal-dialog.md:
```diff
@@ -198,8 +198,41 @@ Some browsers require additional scripting. This simple example works in Chrome,
+  Your browser does not support the video tag.
+</video>
+
+
```
Known issue, I'll remove the extra row when I update the TODO needed in the code example.

tritland commented on public/content/documentation/web/component/modal-dialog.md:
```diff
@@ -6,75 +6,230 @@ How to test a modal dialog
+Full information: [https://www.magentaa11y.com/MagentaA11yV2#/web-criteria/component/modal-dialog](https://www.magentaa11y.com/MagentaA11yV2#/web-criteria/component/modal-dialog)
 
-   - WHEN I use a mobile screenreader (Talkback, VoiceOver) AND
+## Developer Notes
```
Should we remove "Developer Notes" here? There are two h2's right on top of each other

marshcooke replied:
Instead of removing "Developer Notes", which is what that tab is looking for to parse it out, I adjusted the H2 below it and everything else within that section to be the correct structure.

tritland replied:
All good! Your call. I did double check Lindsay's video  and she says (at 16 min, 30 sec) that we don't need to keep "Developer Notes". It should still render everything in the correct tab. But I'm good with whatever you decide here!

tritland commented on public/content/documentation/web/component/modal-dialog.md:
```diff
@@ -6,75 +6,230 @@ How to test a modal dialog
+   - Use `aria-modal="true"` to indicate content beneath the modal is inert and that the screen reader must not browse outside the dialog.
+
+### Focus
+   - Use `tabindex="-1"` to make the modal itself targetable for focus.
```
I know that we do this...but should we consider removing this line about making the modal container focusable? I'm not sure if it's really best practice to do so. Maybe we could do a quick poll?

marshcooke replied:
posed to the team in the MagentaA11Y channel!

marshcooke replied:
I opted to keep this one in for now!

tritland commented on public/content/documentation/web/component/modal-dialog.md:
```diff
@@ -6,75 +6,230 @@ How to test a modal dialog
+
+## Code examples
+
+### Use semantic HTML where possible
```
I also think we should consider dropping all of the <dialog> content. We don't recommend it internally, so why include it here if it's inconsistently supported across browsers. Another poll? ;-)

marshcooke replied:
posed to the team in the MagentaA11Y channel!

tritland commented on public/content/documentation/web/component/modal-dialog.md:
````diff
@@ -6,75 +6,230 @@ How to test a modal dialog
+```
+<!-- TODO: The code sample example is not currently functional - fix needed. -->
+
+<example>
````
not reviewing because it's still broken, right?

marshcooke replied:
Correct, a future todo!

tritland reviewed (changes requested):
Just a few comments!

marshcooke commented:
I changed the heading structure back to what it was, and removed "dev notes". Also kept the references to tabindex="=1", but updated the dialog element and role language.
